design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,187 @@
+* A `Deployment` is created in the same namespace as the stack and it is expected to launch and manage the pod that hosts the GCP controllers.
+
+It is worth reiterating the scope of permissions that this platform stack has received:
+
+> Because the stack is given a `ClusterRole`, it can be granted permissions to cluster-scoped resources and to namespaced resources across all namespaces.
+
+### Architecture Diagram - Platform Stack
+
+The diagram below summarizes the artifacts in the control plane after the GCP stack has been installed and a couple example `MySQLClaims` have been created.
+
+![Diagram of a Platform Stack Installed](./images/platform-stack-installed.png)
+
+### Environments
+
+While platform stacks are installed at the cluster level (there is only one instance of each platform stack), they can still support the concept of multiple "environments", each represented by a namespace.
+For example, an admin can create a namespace to model their production "environment" and another namespace to model their development "environment".
+All resources within each of those environments can share a common set of provider credentials.
+This namespaced "environment" approach provides the administrator with a way to collect or aggregate all the infrastructure and services associated with a given environment.
+
+#### Platform Stack Single Instance Version Limitation
+
+One caveat of this approach is that because there is only a single instance of the platform stack in the cluster, it is therefore shared across all environments.
+This means that all environments must use the same version of the stack.
+If an admin wants to try a newer version of the stack for their dev environment, it will also result in the newer version of the stack being used in the production environment as well.
+
+This limitation could be addressed in the future by allowing multiple versions of platform stacks installed per control plane, which would require additional complexity and design work.
+To get started, we will make the simplifying constraint that only a single version of a platform stack can be installed and it will be shared across all namespaces (environments).
+
+#### Environment Policies
+
+Another potential advantage of supporting multiple namespace "environments" is that each one could be assigned their own specific policy, such as default resource classes.
+The production environment could have a completely different set of default resource classes than the development environment.
+Since these environments are simply represented by namespaces, they can take on any arbitrary meaning or scope.
+For example, instead of dev vs. prod, they could represent entirely other dimensions, such as a high performance environment, a cheap environment, an AWS only environment, etc.
+
+The full design for this support of different policies per environment is out of scope of this document, but this idea has interesting potential.
```

Comment:
Just wanted to call out here that currently a `RedisPolicy` (the only kind implemented so far) must exist in the same namespace as the claim that uses it for defaulting. This could definitely be changed in the future. There are a whole suite of potential features that policies could enable for us.